### PR TITLE
[ALLUXIO-3290] Update UFS mode to journal as a part of modify

### PR DESCRIPTION
**Pull request by aaudiber**
https://alluxio.atlassian.net/browse/ALLUXIO-3290

### CODE REVIEW

File: core/server/master/src/main/java/alluxio/underfs/MasterUfsManager.java [hunk 1]
```diff
@@ -12,137 +12,65 @@
...
+  public MasterUfsManager() {
+    mState = new State();
+    mUfsRoots = new HashSet<>();
+    mIdToRoot = new HashMap<>();
+  }
 
   @Override
-  public void addMount(long mountId, final AlluxioURI ufsUri,
-                       final UnderFileSystemConfiguration ufsConf) {
+  public synchronized void addMount(long mountId, AlluxioURI ufsUri,
```
Comment (gpang) [inline core/server/master/src/main/java/alluxio/underfs/MasterUfsManager.java]: what is the `synchronized` protecting?

File: core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java [hunk 2]
```diff
@@ -394,7 +393,8 @@ public String getName() {
   @Override
   public void processJournalEntry(JournalEntry entry) throws IOException {
     if (mDirectoryIdGenerator.replayJournalEntryFromJournal(entry)
-        || mInodeTree.replayJournalEntryFromJournal(entry)) {
+        || mInodeTree.replayJournalEntryFromJournal(entry)
+        || mUfsManager.replayJournalEntryFromJournal(entry)) {
```
[the remark this answers is not in the document]
Comment (aaudiber) [reply core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java]: Yep: https://github.com/Alluxio/alluxio/blob/master/core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java#L455

File: core/server/master/src/main/java/alluxio/underfs/MasterUfsManager.java (hunk 1 above)
Comment (aaudiber) [reply core/server/master/src/main/java/alluxio/underfs/MasterUfsManager.java]: it protects all the state in the class. I'll added some `GuardedBy` annotations

File: tests/src/main/java/alluxio/master/backcompat/ops/UpdateUfsMode.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    masterClient.updateUfsMode(new AlluxioURI("/"),
+        UpdateUfsModeOptions.defaults().setUfsMode(UfsMode.NO_ACCESS));
+    masterClient.updateUfsMode(new AlluxioURI("/"),
+        UpdateUfsModeOptions.defaults().setUfsMode(UfsMode.READ_WRITE));
+  }
+
+  @Override
+  public void check(Clients clients) throws Exception {
+    FileSystemMasterClient masterClient = clients.getFileSystemMasterClient();
+    assertFalse(masterClient.getMountTable().get("/").getReadOnly());
```
[the remark this answers is not in the document]
Comment (madanadit) [reply tests/src/main/java/alluxio/master/backcompat/ops/UpdateUfsMode.java]: i see, makes sense

**Pull request merged**